Fix bug with ActivelyMicrowavedComponent being stuck in powercuts

DrMelon opened the pull request:
See: #23835, https://github.com/space-wizards/space-station-14/pull/23835#issuecomment-1890011433, https://discord.com/channels/310555209753690112/1195362883928862820

<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
ActivelyMicrowavedComponent is now removed from food inside the microwave when power is cut.

## Why / Balance
Previously, when power was cut, ActivelyMicrowavedComponent would stay on the ingredients - rendering them unusuable forever. 

## Technical details
Two lines of code in the event that responds to power outages.

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
None needed.
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->
None

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- fix: The Microwave works as intended after power outages again.

EmoGarbage404 reviewed (changes requested):
instead of just adding this call into all the logic where items get ejected, add a subscription to `EntRemovedFromContainerMessage` and then remove the component there.

DrMelon commented:
> instead of just adding this call into all the logic where items get ejected, add a subscription to `EntRemovedFromContainerMessage` and then remove the component there.

I didn't add any calls into logic for item ejection - only for the logic where items are _not_ ejected, in the case of a power cut. However, I'll do this refactor.

EmoGarbage404 commented:
> I didn't add any calls into logic for item ejection - only for the logic where items are _not_ ejected, in the case of a power cut. However, I'll do this refactor.

oh i forgot that doesn't happen any more. There should really be a generic StopMicrowaving in that case, that's called both when the microwave finishes and when the microwave loses power.

deathride58 commented:
From our own investigations, it looks like microwaves are incredibly shonky in general when reacting to power outages

DrMelon commented:
Refactored, added a StopCooking() function that specifies if the cooking effort failed or succeeded, so that the components are properly removed on a microwave being broken or unpowered (but retaining successful cooking results if otherwise left alone).

Tin-Man-Tim commented:
So many issues with the microwave, lmao.

deathride58 commented on Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:
```diff
@@ -460,6 +460,32 @@ public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user
 
+        public void StopCooking(Entity<ActiveMicrowaveComponent?, MicrowaveComponent> ent, bool failure)
+        {
+            if (!Resolve(ent, ref ent.Comp1))
```
Tests are currently failing due to this line, as `ActiveMicrowaveComponent` can be null here. It'd probably be better to not pass `ActiveMicrowaveComponent` as an arg, but rather, opt for a `TryComp<ActiveMicrowaveComponent>(etc etc etc)`

DrMelon replied:
Should be fixed. Ran the integration tests locally successfully under Rider 2023.3.

ShadowCommander commented on Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:
```diff
@@ -460,6 +460,32 @@ public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user
+            if (!TryComp<ActiveMicrowaveComponent>(ent, out var activeMicrowave))
+                return;
+
+            RemComp<ActiveMicrowaveComponent>(ent);
```
Move this to later just in case it gets deleted before the other code using its fields get run.

DrMelon replied:
Done.

Nopey commented:
I think chef players would appreciate a changelog entry, unless there are other known bugs that will still break the microwave.
I've seen several chefs give up completely when their microwave stopped cooking food, don't remember if that was on LRP or MRP.

DrMelon commented:
> I think chef players would appreciate a changelog entry, unless there are other known bugs that will still break the microwave. I've seen several chefs give up completely when their microwave stopped cooking food, don't remember if that was on LRP or MRP.

Agreed, added changelog line to PR.
